src/modalities/dataloader/samplers.py

```diff
@@ -1,60 +1,132 @@
-from typing import Optional
+import math
+from typing import Iterator, Optional, TypeVar
 
-from torch.utils.data import BatchSampler, Sampler
+import torch
+import torch.distributed as dist
+from torch.utils.data import Dataset, Sampler
 
+T_co = TypeVar("T_co", covariant=True)
+
+
+class ResumableDistributedSampler(Sampler[T_co]):
```

Comment:
I can't remember why we decided for adapting DistributedSampler in pytorch instead of inheriting or using composition. From the start of modalities, wie basically used the adaptation.  We could make this another issue to investigate if it makes sense to move to the original pytorch DistributedSampler.